google/resource_bigtable_instance.go

```diff
@@ -0,0 +1,204 @@
+			},
+
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceBigtableInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	displayName, ok := d.GetOk("display_name")
+	if !ok {
+		displayName = name
+	}
+
+	var storageType bigtable.StorageType
+	switch value := d.Get("storage_type"); value {
+	case "HDD":
+		storageType = bigtable.HDD
+	case "SSD":
+		storageType = bigtable.SSD
+	}
+
+	instanceConf := &bigtable.InstanceConf{
+		InstanceId:  name,
+		DisplayName: displayName.(string),
+		ClusterId:   d.Get("cluster_id").(string),
+		NumNodes:    int32(d.Get("num_nodes").(int)),
+		StorageType: storageType,
+		Zone:        d.Get("zone").(string),
+	}
+
+	c, err := config.bigtableClientFactory.NewInstanceAdminClient(project)
+	if err != nil {
+		return fmt.Errorf("Error starting instance admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	err = c.CreateInstance(ctx, instanceConf)
+	if err != nil {
+		return fmt.Errorf("Error creating instance. %s", err)
+	}
+
+	d.SetId(name)
+
+	return resourceBigtableInstanceRead(d, meta)
+}
+
+func resourceBigtableInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	c, err := config.bigtableClientFactory.NewInstanceAdminClient(project)
+	if err != nil {
+		return fmt.Errorf("Error starting instance admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	instances, err := c.Instances(ctx)
+	if err != nil {
+		return fmt.Errorf("Error retrieving instances. %s", err)
+	}
+
+	var instanceInfo *bigtable.InstanceInfo
+	name := d.Id()
+	found := false
+	for _, i := range instances {
+		if i.Name == name {
+			instanceInfo = i
+			found = true
+			break
+		}
+	}
+	if !found {
+		return fmt.Errorf("Error retrieving instance. Could not find %s.", name)
+	}
+
+	d.Set("name", instanceInfo.Name)
+	d.Set("display_name", instanceInfo.DisplayName)
+
+	return nil
+}
+
+func resourceBigtableInstanceDestroy(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	c, err := config.bigtableClientFactory.NewInstanceAdminClient(project)
+	if err != nil {
+		return fmt.Errorf("Error starting instance admin client. %s", err)
+	}
+
+	defer c.Close()
+
+	name := d.Id()
+	err = c.DeleteInstance(ctx, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting instance. %s", err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+// IntAtLeast returns a SchemaValidateFunc which tests if the provided value
```

Comment:
Yep! I was just working on that after replying to the comments on this.

https://github.com/hashicorp/terraform/pull/15403